### PR TITLE
CI: move `pytest` configuration to `pyproject.toml`

### PR DESCRIPTION
**Pull request by sphuber**
Part of the `pytest` configuration was done through environment
variables in the `.github/workflows/tests.sh`. This is moved to the
`pyproject.toml` where the rest of the configuration is and as such
everything is one place.

The `.coveragerc` configuration file no longer seems necessary since the
configuration is specified directly.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -57,7 +57,7 @@ max-locals = 20
 
 [tool.pytest.ini_options]
 minversion = "6.0"
-addopts = "--benchmark-skip"
+addopts = "--benchmark-skip --durations=50 --cov aiida --cov-report xml --cov-append --verbose"
```
Comment (chrisjsewell) [inline pyproject.toml:60]: The only problem is that it forces every run of pytest to be verbose now and output all the coverage, i.e. when I'm testing locally.
At least for `verbose`, I would rather have this set in the workflow job

Comment (sphuber) [reply pyproject.toml:60]: Ah yeah, that's true. Maybe I move `--cov aiida` and `--verbose` to the actual invocation in `tests.sh`.

**Comment by sphuber** [comment]
@chrisjsewell want to have a final look at this one?

**Review by chrisjsewell — APPROVED** [review]
Yeh should be fine cheers

**Pull request merged**